include/bbp/sonata/report_reader.h

```diff
@@ -123,19 +123,22 @@ class SONATA_API ReportReader
 
         /**
          * \param node_ids limit the report to the given selection.
-         * \param tstart return spikes occurring on or after tstart. tstart=nonstd::nullopt
-         * indicates no limit. \param tstop return spikes occurring on or before tstop.
-         * tstop=nonstd::nullopt indicates no limit.
+         * \param tstart return voltages occurring on or after tstart. tstart=nonstd::nullopt
+         * indicates no limit. \param tstop return voltages occurring on or before tstop.
+         * tstop=nonstd::nullopt indicates no limit. \param tstride indicates every how many
+         * timesteps we read data. tstride=nonstd::nullopt indicates that all timesteps are read.
          */
         DataFrame<KeyType> get(const nonstd::optional<Selection>& node_ids = nonstd::nullopt,
                                const nonstd::optional<double>& tstart = nonstd::nullopt,
-                               const nonstd::optional<double>& tstop = nonstd::nullopt) const;
+                               const nonstd::optional<double>& tstop = nonstd::nullopt,
+                               const nonstd::optional<size_t>& tstride = nonstd::nullopt) const;
 
       private:
         Population(const H5::File& file, const std::string& populationName);
-        std::pair<size_t, size_t> getIndex(const nonstd::optional<double>& tstart, const nonstd::optional<double>& tstop) const;
+        std::pair<size_t, size_t> getIndex(const nonstd::optional<double>& tstart,
+                                           const nonstd::optional<double>& tstop) const;
 
-        std::vector<std::pair<NodeID, std::pair<uint64_t, uint64_t>>> nodes_pointers_;
+        std::map<NodeID, std::pair<uint64_t, uint64_t>> nodes_pointers_;
```

Comment:
The idea is to use the black red tree to do fast search inside NodeID.
NodeID can be 2 millions and we need to select maybe 10,000 ids.
They are not sorted.

`nodes_pointers_` is a sorted cache

Reply:
Exactly!

Reply:
I'm wondering: shouldn't the performance of `unordered_map` (unordered, O(1) access time complexity) and sorting the results only once and only when all the nodes are selected (i.e. `selection` is not specified when calling `get`) be better than using `map` (ordered, O(log N) access time complexity)?
Anyway the memory usage would be higher.

Reply:
The point is that, while the unordered map can technically provide certain advantages, a higher amount of GIDs can also increase the chances for conflicts in the buckets of the hash table. This means that it could potentially be a sequential O(N) search of certain elements.

Nonetheless, we tried to use first an unordered map and the performance was about the same as a conventional map. Thus, we kept the conventional map to guarantee the same access time and lower the memory footprint. In addition, right now the search for the GIDs amounts for a very little piece of the overall execution time, even for 100K elements.

Reply:
OK, was curious… I guess in the end IO is probably going to overwhelm any performance tuning here, so KISS is better. Thanks for the follow-up!